app/models/restaurant.rb

```diff
@@ -5,11 +5,24 @@ class Restaurant < ApplicationRecord
   validates :state, presence: true
   validates :zip,
     numericality: true,
-    presence: true,
     length: { is: 5 }
   validates :phone_number,
     presence: true,
     format: { with: /\d{3}-\d{3}-\d{4}/ }
+  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i }
+
+  before_validation :strip_whitespace
+
+  def strip_whitespace
+    self.website = self.website.strip unless self.website.nil?
+    self.email = self.email.strip unless self.email.nil?
+    self.phone_number = self.phone_number.strip unless self.phone_number.nil?
+    self.name = self.name.strip unless self.name.nil?
+    self.street = self.street.strip unless self.street.nil?
+    self.city = self.city.strip unless self.city.nil?
+    self.state = self.state.strip unless self.state.nil?
+  end
```

Comment:
Interesting. If you are going to jump through this hoop, You will have to test your method as well 

https://karolgalanciak.com/blog/2017/10/29/the-case-against-exotic-usage-of-before-validate-callbacks/